src/Utils/RMC/RMST.as

```diff
@@ -219,25 +208,26 @@ class RMTS : RMS {
         MXNadeoServicesGlobal::ClubRoomSetMapAndSwitchAsync(RMTRoom, currentMap.TrackUID);
     }
 
-    void SetServerTime() {
-        // set the server time to have a little bit more time to make sure the plugin catches up
-        uint ServerRemainingTime = (RunRemainingTime() / 1000) + 5;
-        MXNadeoServicesGlobal::ClubRoomSetCountdownTimer(RMTRoom, ServerRemainingTime);
-    }
-
     void ResetToLobbyMap() {
         if (LobbyMapUID != "") {
             UI::ShowNotification("Returning to lobby map", "Please wait...", Text::ParseHexColor("#993f03"));
-            BetterChatSendMessage(Icons::Users + " Returning to lobby map...");
+            BetterChatSendMessage(Icons::Scuttlebutt + " Returning to lobby map...");
             MXNadeoServicesGlobal::SetMapToClubRoomAsync(RMTRoom, LobbyMapUID);
-            if (UserEndedRun) MXNadeoServicesGlobal::ClubRoomSwitchMapAsync(RMTRoom);
+            if (RMC::UserEndedRun) MXNadeoServicesGlobal::ClubRoomSwitchMapAsync(RMTRoom);
             while (!TM::IsMapCorrect(LobbyMapUID)) sleep(500);
-            UserEndedRun = false;
+            RMC::UserEndedRun = false;
         }
-        ResetValues();
         MXNadeoServicesGlobal::ClubRoomSetCountdownTimer(RMTRoom, 0);
     }
 
+    void PendingTimerLoop() override
+    {
+        // Cap timer max
+        if ((RMC::EndTime - RMC::StartTime) > (PluginSettings::RMC_SurvivalMaxTime-RMC::SurvivalTogether.Skips)*60*1000) {
+            RMC::EndTime = RMC::StartTime + (PluginSettings::RMC_SurvivalMaxTime-RMC::SurvivalTogether.Skips)*60*1000;
```

Comment:
I don't know the exact data type here (i.e., I don't know if AngelScript supports `auto`):

```suggestion
        const auto MAX_TIME = (PluginSettings::RMC_SurvivalMaxTime - RMC::SurvivalTogether.Skips)
            * 60 * 1000;
        if ((RMC::EndTime - RMC::StartTime) > MAX_TIME {
            RMC::EndTime = RMC::StartTime + MAX_TIME;
```